wordpress-mobile/WordPress-iOS
# Issues/reader post sync

This PR updates the way the reader syncs content, improving the way gaps are detected and resolved at the cost of small amount of data overhead.

Gaps are flagged by a "gap marker" in core data. 
Gap marker's are represented in the post list by a custom cell displaying a "Load more posts" button. 
Tapping the button syncs the missing posts, discarding the cached posts below the gap.

To test:
Choose a high traffic feed like "art", and to edit the value of `ReaderPostServiceNumberToSync` to `5` or so.  Watch the same feed in the web reader and note when there are more new posts available than the number you are syncing.  On a busy feed this should only take a couple of minutes. 

Scenario 1
- Pull to refresh and confirm that the gap marker cell appears. 
- Tap the cell should show a highlight, and a spinner. 
- Confirm the missing content is loaded and the gap cell is no longer displayed.

Scenario 2
- Pull to refresh and confirm that a gap marker appears. 
- Wait a bit for new content to become available but not so much that there would be another gap.
- Pull to refresh again. 
- Ensure the new content appears, and the gap marker for the previous content remains. 

Scenario 3
- Pull to refresh and confirm that a gap marker appears. 
- Wait a bit for new content to become available, but this time make sure there is enough to create a new gap
- Pull to refresh again. 
- Ensure the new content appears. 
- Ensure a gap marker appears below the newly synced content. 
- Ensure the previous gap marker is no longer shown, as we'll only support filling a single gap in content (since older posts will be discarded anyway).

cc @shaunandrews.  (Note these early screenshots show a 4px margin on either side of the button but a 2px was used in the end.)

Screen shots:
![simulator screen shot sep 25 2015 10 42 54 am](https://cloud.githubusercontent.com/assets/1435271/10106123/7288efa2-6378-11e5-8201-3ead687069a4.png)

![simulator screen shot sep 25 2015 10 42 56 am](https://cloud.githubusercontent.com/assets/1435271/10106129/7a2f9076-6378-11e5-88f2-a473bdf4bb34.png)

![simulator screen shot sep 25 2015 10 42 57 am](https://cloud.githubusercontent.com/assets/1435271/10106136/801beb7e-6378-11e5-87c3-0031472869bc.png)

@jleandroperez Could I trouble you for a review on this one? 

Needs review: @jleandroperez
